{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -6,136 +6,159 @@
 render model specs with the correct location parameters plugged in.
 
 It will look for the model spec template in "model_spec.in" in the directory
-"../{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/model_specification".
-Specify multiple locations in a file called "locations.txt" in the same directory.
-You can create a single location using the "-s" option. Using the single
-location option will override an existing locations file.
+``src/{{cookiecutter.package_name}}/model_specifications``.
+Specify multiple locations in a file called "locations.txt" in the same
+directory. You can create a single location using the "-s" option. Using the
+single location option will override an existing locations file.
 
 The application will look for the model spec and locations files based
 on the python environment that is active and these files don't need
 to be specified if the default names and location are used.
 """
+from collections import namedtuple
+from pathlib import Path
 import re
+from typing import List, Optional, Generator
+
 import click
-from pathlib import Path
 from jinja2 import Template
-from collections import namedtuple
 from loguru import logger
-from typing import List, Callable
-from functools import wraps
-from sys import exit
 
 
-PROJ_NAME = '{{cookiecutter.package_name}}'
-Location = namedtuple('Location', ['loc_proper', 'loc_sanitized'])
+PROJECT_NAME = '{{cookiecutter.package_name}}'
+Location = namedtuple('Location', ['proper', 'sanitized'])
+
+
+def sanitize(*locations: str) -> Generator[Location]:
+    """Processes locations into tuples of proper and sanitized names.
```

Comment:
We use the numpy docstring format: https://numpydoc.readthedocs.io/en/latest/format.html with the only major difference that we put typing annotations in the function sigs instead of the docstrings themselves.

Important things: Docstrings are parsed as restructured text in all of our documentation generating tools.  You'll see liberal usage of sphinx directives and other things in the framework code, and we're working to expand that out into the rest of the codebase.  These docs probably won't be generated and hosted for a while (though I do think having docs pages for the research repos is a good idea), due to, well, time and resource constraints.

I try to be consistent at wrapping docstrings at 79 characters so they display properly in terminals, though you'll see we're a bit inconsistent about that in other places in the code base.